### PR TITLE
Refactor migrations to avoid stored procedure/schema mismatches

### PR DESCRIPTION
**Pull request by apmasell**
To migrate on prod, first do:
```
UPDATE schema_version SET checksum = -984436609 WHERE version='0163';
DELETE FROM schema_version WHERE version='0164';
```

### CODE REVIEW

File: sqlstore/src/main/resources/db/migration_afterMigrate/70-kits.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- StartNoTest
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration_afterMigrate/70-kits.sql]: FYI, you've got a dash after the number in the file name here, and an underscore after the number elsewhere.

Comment (apmasell) [reply sqlstore/src/main/resources/db/migration_afterMigrate/70-kits.sql]: Fixed.

File: sqlstore/src/main/resources/db/migration/V0163__auto_site.sql [hunk 2]
```diff
@@ -1,13 +1,7 @@
...
---StartNoTest
-CALL addTargetedSequencing('Agilent SureSelectXT MethylSeq', 'TGL', 'Agilent SureSelectXT MethylSeq', 0);
-CALL addTargetedSequencing('IDT xGEN Exome Research Panel v1', 'TGL', 'IDT xGEN Exome Research Panel v1', 0);
-CALL addTargetedSequencing('TruSeq RNA Access', 'TGL', 'TruSeq RNA Access', 0);
-CALL addTargetedSequencing('Roche SeqCap Epi CpGiant', 'TGL', 'Roche SeqCap Epi CpGiant', 0);
-CALL addTargetedSequencing('TruSeq Methylation Epic Exome', 'TGL', 'TruSeq Methylation Epic Exome', 0);
-
-CALL addKitDescriptor('Agilent SureSelect Human All Exon V5 + UTRs', 1, 'Agilent', 1, 'Library', 'Illumina', 'TGL');
+INSERT INTO KitDescriptor(name, version, manufacturer, partNumber, kitType, platformType, description, lastModifier)
+  SELECT 'Agilent SureSelect Human All Exon V5 + UTRs', 1, 'Agilent', 1, 'Library', 'Illumina', 'TGL', getAdminUserId() FROM DUAL
```
Comment (djcooke) [inline sqlstore/src/main/resources/db/migration/V0163__auto_site.sql]: `getAdminUserId` is a function that's also created in beforeMigrate, so shouldn't be used in regular migrations for the same reason as the procedures

Comment (apmasell) [reply sqlstore/src/main/resources/db/migration/V0163__auto_site.sql]: If we can guarantee it will exist, I'm willing to believe the API is stable, no?

Comment (djcooke) [reply sqlstore/src/main/resources/db/migration/V0163__auto_site.sql]: It's stable for now, but avoiding the function would be a small change and guarantee that it never breaks

File: sqlstore/src/main/resources/db/migration_afterMigrate/90_targeted_sequencing.sql [hunk 3]
```diff
@@ -1,66 +1,69 @@
...
 CALL addTargetedSequencing('SureSelect v5','GSLE Legacy','KAPA Hyper Prep',1);
-CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','KAPA Hyper Prep',1);
-CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','Kapa Hyper Prep Plus',1);
-CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','KAPA OnBead ',1);
-CALL addTargetedSequencing('TruSeq RNA Access RS-301-2001','GSLE Legacy','TruSeq RNA Access ',0);
-CALL addTargetedSequencing('DREAM_IAD60445','GSLE Legacy','AmpliSeq-Illumina V1',0);
-CALL addTargetedSequencing('HALO_IBP','GSLE Legacy','HALO',0);
-CALL addTargetedSequencing('Nextera Rapid Capture Expanded Exome','GSLE Legacy','Nextera XT',0);
-CALL addKitDescriptor('QXT',0,'Agilent',0,'LIBRARY','ILLUMINA',NULL);
 CALL addTargetedSequencing('SureSelect v5','GSLE Legacy','QXT',0);
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration_afterMigrate/90_targeted_sequencing.sql:57]: Migration failure: KitDescriptor 'QXT' not found. Final lines from generated `afterMigrate` file:
```CALL addTargetedSequencing('TruSeq RNA Access RS-301-2001','GSLE Legacy','TruSeq RNA Access ',0);
CALL addTargetedSequencing('TruSeq RNA Access', 'TGL', 'TruSeq RNA Access', 0);
CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','GA_PE',1);
CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','KAPA BC OnBead',1);
CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','KAPA Hyper Prep',1);
CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','KAPA OnBead ',1);
CALL addTargetedSequencing('xGen Lockdown AML','GSLE Legacy','Kapa Hyper Prep Plus',1);
-- EndNoTest
-- StartNoTest
CALL addKitDescriptor('QXT',0,'Agilent',0,'LIBRARY','ILLUMINA',NULL);
-- EndNoTest
-- StartNoTest
CALL addInstitute('Royal Ontario Museum');
CALL addLab('Not Specified', 'Royal Ontario Museum');

CALL addInstitute('Sheba Medical Centre');
CALL addLab('Not Specified', 'Sheba Medical Centre');
CALL addLab('BioBank', 'Sheba Medical Centre');
-- EndNoTest
```

I'd guess the Java is not sorting the afterMigrate files before merging them.

Comment (apmasell) [reply sqlstore/src/main/resources/db/migration_afterMigrate/90_targeted_sequencing.sql:57]: Fixed in https://github.com/TGAC/miso-lims/pull/743

File: sqlstore/src/main/resources/db/migration/V0163__auto_site.sql (hunk 2 above)
Comment (apmasell) [reply sqlstore/src/main/resources/db/migration/V0163__auto_site.sql]: Fixed.

**Comment by callunity** [comment]
Please add the instructions for migrating the prod db to the Release Procedure page.

**Comment by apmasell** [comment]
Added.

**Review by callunity — APPROVED** [review]

**Review by djcooke — APPROVED** [review]

**Pull request merged**